src/server.rs

```diff
@@ -71,6 +71,7 @@ pub fn handle_connection<M: Messenger, PP: PacketProcessor, F: Fn()>(
             Err(e) => {
                 match e.kind() {
                     UnexpectedEof => on_closure(),
+                    ConnectionReset => on_closure(),
```

Comment:
Clients properly send us an EOF, whereas our servers just close connections. So either of these errors isn't an actual error and should trigger the on_closure behaviour